calicoctl/commands/node/diags.go

```diff
@@ -155,7 +160,43 @@ such as transfer.sh using curl or similar.  For example:
 
 }
 
-// writeDiags executes the dignostic commans and outputs the result in the file
+// getDockerLogs will attempt to grab logs for any "calico" named containers for hosted installs.
+func getDockerLogs(logDir string) {
+	os.Mkdir(logDir, os.ModeDir)
+
+	fmt.Println("Copying logs from Calico containers")
+	// Get a list of Calico containers running on this Node.
+	var stderr bytes.Buffer
+	cmd := exec.Command("docker", "ps", "-a", "--filter", "\"name=calico\"", "--format", "{{.Names}}")
+	cmd.Stderr = &stderr
```

Comment:
This gives me a runtime error if `docker` doesn't exist, but works if docker throws an error.  Whereas the other method works in both situations.